pkg/storage/sst_writer.go

```diff
@@ -57,6 +57,34 @@ func (noopSyncCloser) Close() error {
 	return nil
 }
 
+type flusher interface {
+	Flush() error
+}
+
+// noopSyncCloserFlusher is like noopSyncCloser but also implements the flusher
+// interface.
+type noopSyncCloserFlusher struct {
+	io.Writer
+}
+
+func (noopSyncCloserFlusher) Sync() error {
+	return nil
+}
+
+func (noopSyncCloserFlusher) Close() error {
+	return nil
+}
+
+func (n noopSyncCloserFlusher) Flush() error {
+	f, ok := n.Writer.(flusher)
```

Comment:
you _could_ have two fields in your wrapper -- one writer and one flusher, passing the same _value_ to each at creation where you already have typed it as flusher, and then don't need to re-type it here. I donno if you like that any better than this extra round of casting